packages/terra-table/src/terra-dev-site/doc/table/reference-proptypes/sectionHeaderShape.jsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+/**
+ * The props table parser continues to have trouble detecting the propTypes of a component that
+ * returns `null`. I've duplicated the propType definition here so that a table could be parsed for it.
+ */
+const propTypes = {
+  /**
+   * The react key to apply to the section header.
+   */
+  key: PropTypes.string.isRequired,
+  /**
+   * The id to apply to the header in order to provide structure for AT.
+   */
+  id: PropTypes.string.isRequired,
+  /**
+  * Whether or not the section is collapsed.
+  */
+  isCollapsed: PropTypes.bool,
+  /**
+  * Whether or not the section can be collapsed.
+  */
+  isCollapsible: PropTypes.bool,
+  /**
+  * The associated metaData to be provided in the onSelect callback.
+  */
+  // eslint-disable-next-line react/forbid-prop-types
+  metaData: PropTypes.object,
+  /**
+   * The numberOfColumns to be used as a descriptor for assistive technology.
+   */
+  numberOfColumns: PropTypes.number.isRequired,
```

Comment:
Can this be pulled from the Header's configuration? This seems like this could be configured at the top level with how the new table API is so far